### DISCUSSION TITLE
Does meshoptimizer preserve UVs? And can it preserve UVs after simplifying polycount of a mesh with gltfpack?

### DISCUSSION THREAD

**Discussion by tdbe** [zeux/meshoptimizer#334] [Q&A] [answered]
This is not explained anywhere in the readme. 
Says it optimizes the mesh and it looks the same etc.

I run a simple gltf to gltf with default settings no extra arguments.
I open it in blender.
Original has texture, optimized does not.

Is this correct / expected? Or am I doing smth wrong?
Ideally I was hoping I can even simplify polycount while preserving UVs. But I guess I'm in the wrong project.

**Comment by zeux** [comment, marked as answer]
The answer to both of these is "yes", although the UV post-simplification - while generally preserved correctly - can sometimes result in UV distortion for complex models, which is pending improvement in the future.

I'd recommend using https://gltf-viewer.donmccurdy.com/ to compare your glTF files before and after gltfpack. If they look identical - which they should - the problem lies with Blender importer. I'm not sure on what extensions it supports, but gltfpack uses a couple KHR extensions by default as noted in the readme.

**Reply by tdbe** [reply]
thanks! you're right, it was a combination of weird mesh + weird blender. It works. It gets artifacts on the topology sometimes when simplifying meshes (dark overlapping / flipped triangles) but that is a different story entirely, so thanks!!
